app/grandchallenge/reader_studies/views.py

```diff
@@ -1393,13 +1393,11 @@ def form_valid(self, form):
 
             if ci.kind in InterfaceKind.interface_type_json():
                 if current_value:
-                    current_value.value = civ
-                    current_value.save()
-                else:
-                    val = ComponentInterfaceValue.objects.create(
-                        interface=ci, value=civ
-                    )
-                    instance.values.add(val)
+                    assigned_civs.append(current_value)
+                val = ComponentInterfaceValue.objects.create(
+                    interface=ci, value=civ
+                )
+                instance.values.add(val)
```

Comment:
I have also made some changes to this logic in #2575, but I'll be sure to check those if this PR gets merged first.

Reply:
Oh right! I see that we actually made the same changes. So that should be easy to integrate if this goes through. I thought I had checked your PR, but clearly missed those changes. Would it make more sense to merge this PR into your branch?

Reply:
Let's keep them separate and I'll fix any conflicts if they arise.